### PR TITLE
[ticket/16696] Fix unsupported operand types in viewforum - PHP8

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16696

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

https://tracker.phpbb.com/browse/PHPBB3-16696

### CODE REVIEW

File: phpBB/viewforum.php [hunk 1]
```diff
@@ -1027,7 +1027,7 @@
 
 		$template->assign_block_vars('topicrow', $topic_row);
 
-		$pagination->generate_template_pagination($topic_row['U_VIEW_TOPIC'], 'topicrow.pagination', 'start', $topic_row['REPLIES'] + 1, $config['posts_per_page'], 1, true, true);
+		$pagination->generate_template_pagination($topic_row['U_VIEW_TOPIC'], 'topicrow.pagination', 'start', (int) $topic_row['REPLIES'] + 1, $config['posts_per_page'], 1, true, true);
```
[the remark this answers is not in the document]
Comment (3D-I) [reply phpBB/viewforum.php:1030]: Is related to extensions that use the `core.viewforum_modify_topicrow` event.

[the remark this answers is not in the document]
Comment (CHItA) [reply phpBB/viewforum.php:1030]: I've seen that. My question is why is this necessary? I would have assumed `+` implicitly casts anyway. Is that not the case?

[the remark this answers is not in the document]
Comment (3D-I) [reply phpBB/viewforum.php:1030]: No, is not the case. It breaks in php 7 and PHP 8 is more explicit.

I will provide a code example where you can see it in action later on.
The real issue is where the PR which caused this BC regression has been merged, anyway this simple fix solves the issue.

[the remark this answers is not in the document]
Comment (marc1706) [reply phpBB/viewforum.php:1030]: For context: https://www.phpbb.com/community/viewtopic.php?f=461&t=2582631

[the remark this answers is not in the document]
Comment (3D-I) [reply phpBB/viewforum.php:1030]: Moreover, there are potentially a lot of extensions which can break via that PR.
https://github.com/search?q=core.viewforum_modify_topicrow&type=code
My first impression it was to revert that PR.

[the remark this answers is not in the document]
Comment (marc1706) [reply phpBB/viewforum.php:1030]: The problem is that extension authors have been using a value that was initially an integer value to change it to strings. As a result of that, casting to int would get rid of the warning but still somewhat has a potential of ending up with weird output. Anyway, weird output is still better than PHP warnings being output.

**Pull request merged**